src/parquet/encodings/dictionary-encoding.h

```diff
@@ -418,6 +432,23 @@ inline void DictEncoder<FixedLenByteArray>::WriteDict(uint8_t* buffer) {
   }
 }
 
+template<typename T>
+inline void DictEncoder<T>::FreeMemory() {}
+
+template<>
+inline void DictEncoder<ByteArray>::FreeMemory() {
+  for (const auto& value : uniques_) {
+    pool_->Free(const_cast<uint8_t*>(value.ptr));
+  }
+}
+
+template<>
+inline void DictEncoder<FixedLenByteArray>::FreeMemory() {
+  for (const auto& value : uniques_) {
+    pool_->Free(const_cast<uint8_t*>(value.ptr));
+  }
+}
```

Comment:
This code is concerning given that we don't have performance benchmarks. In the prior version, we were requesting memory from a chunked allocator, then `FreeAll` was deallocating chunks. Here, if we have a lot of small strings, all of these small frees (virtual function calls, plus `std::free`) are likely to have much worse performance.